pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -153,8 +154,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 
 		if err := r.createOrUpdateExternalSecret(ctx, &clusterExternalSecret, namespace, esName, clusterExternalSecret.Spec.ExternalSecretMetadata); err != nil {
-			log.Error(err, "failed to create or update external secret")
-			failedNamespaces[namespace.Name] = err
+			if !ctrlerrors.IsNamespaceGone(err) {
+				log.Error(err, "failed to create or update external secret")
+				failedNamespaces[namespace.Name] = err
+			}
```

Comment:
nit: maybe it makes sense that if we do have gone namespace, we log a a debug message